ide/static/ide/js/editor.js

```diff
@@ -67,10 +67,26 @@ CloudPebble.Editor = (function() {
 
         // Open it.
         return Ajax.Get('/ide/project/' + PROJECT_ID + '/source/' + file.id + '/load').then(function(data) {
-            var is_js = file.name.substr(-3) == '.js';
             var source = data.source;
             file.lastModified = data.modified;
             var pane = $('<div>');
+            var file_kind, file_mode;
+            if (/\.js$/.test(file.name)) {
+                file_kind = 'js';
+                file_mode = 'javascript';
+            }
+            else if (/\.json$/.test(file.name)) {
+                file_kind = 'json';
+                file_mode = 'application/json';
+            }
+            else {
+                file_kind = 'c';
+                file_mode = CloudPebble.Editor.PebbleMode;
+            }
+            function file_kind_in(options) {
+                return _.contains(options.split(' '), file_kind);
```

Comment:
I'd rather this took an array, rather than a string, just on design cleanliness (and it's only called twice anyway…).